packages/data/src/test/registry.js

```diff
@@ -266,27 +262,6 @@ describe( 'createRegistry', () => {
 			return promise;
 		} );
 
-		it( 'should resolve promise action to dispatch', () => {
-			registry.registerReducer( 'demo', ( state = 'NOTOK', action ) => {
-				return action.type === 'SET_OK' ? 'OK' : state;
-			} );
-			registry.registerSelectors( 'demo', {
-				getValue: ( state ) => state,
-			} );
-			registry.registerResolvers( 'demo', {
-				getValue: () => Promise.resolve( { type: 'SET_OK' } ),
```

Comment:
It's not possible to support this usecase in sync generators. Because of this:

imagine the promise returns an object that's not an action, and that object has a "type" property (we have this use case in posts, if we try to fetch a post, we'd receive a promise of an object with a type property but it's not an action), there's no way to know whether we should return this object or dispatch it instead.

In general `redux-saga` and `redux-rungen` solve this issue by requiring sync generators to wrap action inside helpers to avoid ambiguity:

```
function *() {
   yield Promise.resolve( put( { type: 'SET_OK' } ) );
}
```

The `put` helper here doesn't do anything special aside wrapping the given action and an object that can't be ambiguous. We're certain it's an action, so we dispatch the action and not return the object as the yielded result.

Reply:
For what it's worth: getting used to doing that was very easy when we moved to using `redux-saga` on the addons-frontend.